src/main/java/arthur/TaskList.java

```diff
@@ -0,0 +1,205 @@
+        Event temp;
+        try {
+            String[] tempArr = e.split("/at ");
+            temp = new Event(tempArr[0], tempArr[1]);
+        } catch (IndexOutOfBoundsException a) {
+            return "Please add the event Date and/or Time";
+        }
+
+        taskList.add(temp);
+        return "Added a new Task.Event task: \n" + temp
+                + "\n" + this.outstanding();
+    }
+
+    /**
+     * Generates a string about the number of tasks left in list
+     * @return String info regarding num of tasks left
+     */
+    public String outstanding() {
+        return "You have "
+                + this.taskList.size()
+                + " tasks in list at the moment.";
+    }
+
+    /**
+     * Deletes the task from the list.
+     * @param i The task number to delete
+     * @return String conformation of the task deletion
+     */
+    public String deleter(int i) {
+        try {
+            Task currTask = taskList.remove(i - 1);
+            return "Successfully removed this task: \n" + currTask
+                    + "\n" + outstanding();
+        } catch (IndexOutOfBoundsException e) {
+            return "Invalid task number. Please try again";
+        }
+    }
+
+    /**
+     * Copies over tasks in the data file to the taskList.
+     */
+    private void taskFiller() {
+        try {
+            Scanner sc = new Scanner(storage.getTasks());
+            while (sc.hasNext()) {
+                String temp = sc.nextLine();
+                String taskInfo = temp.split(" >> ")[1];
+                char taskType = temp.charAt(1);
+                char marking = temp.charAt(4);
+                switch (taskType) {
+                case 'T':
+                    todo(taskInfo);
+                    break;
+                case 'D':
+                    taskInfo = taskInfo.replaceFirst("\\(By:", "/by");
+                    taskInfo = taskInfo.replaceFirst("\\)", "");
+                    deadline(taskInfo);
+                    break;
+                case 'E':
+                    taskInfo = taskInfo.replaceFirst("\\(At:", "/at");
+                    taskInfo = taskInfo.replaceFirst("\\)", "");
+                    event(taskInfo);
+                    break;
+                }
+
+                if (marking == 'X') {
+                    taskList.get(taskList.size() - 1).mark();
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Gets the Task.Task object at the given index from taskList.
+     * @param num The index of the Task.Task object to retrieve
+     * @return A Task.Task object
+     * @throws IndexOutOfBoundsException Incorrect index needs to be handled properly
+     */
+    public Task getTask(int num) throws IndexOutOfBoundsException {
+        return this.taskList.get(num);
+    }
+
+    /**
+     * Gives the taskList size.
+     * @return list size in int
+     */
+    public int tasksSize() {

```

Comment:
You can add another verb in front of the method name so that it easier to follow, eg: `getTaskListSize`